### PR TITLE
Resurrection counter

### PR DESCRIPTION
**Pull request by sensorsasha**
### Feature

This PR adds a resurrection counter for each resurrection event. It increases a counter each time a job is resurrected. Counter key contains job's `jid`. 

On each resurrection event, a counter is increased. Counter value is stored in redis and has expiration time set to 
24 hours. For example this can be used in the `ServerMiddleware` later on to kill resurrected jobs 
if needed based on the count.

`resurrection_counter` can be both a `Proc` and a constant value. 

Having a `Proc` is useful if you want to enable or disable resurrection counter in run time. It will be called on each 
resurrection event to decide whether to increase the counter or not.

```ruby
Sidekiq::Ultimate.setup! do |config|
  config.enable_resurrection_counter = -> do
    (Time.now.to_i % 2).zero?
  end
end

Sidekiq::Ultimate.setup! do |config|
  config.enable_resurrection_counter = true
end
```

Resurrection counter value can be read using `Sidekiq::Ultimate::Resurrector::Count.read` method.

```ruby
Sidekiq::Ultimate::Resurrector::Count.read(:job_id => "2647c4fe13acc692326bd4c2")
=> 1
```

### Other

- Fix namespacing for LUA scripts by bamping `redis-prescription` gem version
- Migrate to `redlock` from `redis-lockers` because `redis-lockers` does not support newer `redis-prescription`
- Lock `redis-namespace` version to support newer `redis-rb` versions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +54,38 @@ Sidekiq::Ultimate.setup! do |config|
...
+if needed based on the count.
+
+`resurrection_counter` can be both a `Proc` and a constant value. 
+
+Having a `Proc` is useful if you want to enable or disable resurrection counter in run time. It will be called on each 
+resurrection event to decide whether to increase the counter or not.
+
+```ruby
+Sidekiq::Ultimate.setup! do |config|
+  config.resurrection_counter = -> do
```
Comment (DenisSychyov) [inline README.md]: Let's provide more _practical_ example

File: lib/sidekiq/ultimate/configuration.rb [hunk 2]
```diff
@@ -16,7 +16,7 @@ class Configuration
       #  Sidekiq::Ultimate::Configuration.instance.on_resurrection = ->(queue_name, jobs_count) do
       #    puts "Resurrected #{jobs_count} jobs from #{queue_name}"
       #  end
-      attr_accessor :on_resurrection
+      attr_accessor :on_resurrection, :enable_resurrection_counter
```
Comment (DenisSychyov) [inline lib/sidekiq/ultimate/configuration.rb:19]: Should it be `resurrection_counter`?

File: spec/sidekiq/ultimate/resurrector_spec.rb [hunk 3]
```diff
@@ -57,7 +92,60 @@
...
+        counter2 = Sidekiq.redis { |r| r.get("ultimate:resurrector:counter:jid:2647c4fe13acc692326bd4c3") }
+        counter2_ttl = Sidekiq.redis { |r| r.ttl("ultimate:resurrector:counter:jid:2647c4fe13acc692326bd4c3") }
+
+        expect(counter1).to eq("1")
+        expect(counter1_ttl).to be_within(5).of(86_400)
+        expect(counter2).to eq("1")
+        expect(counter2_ttl).to be_within(5).of(86_400)
+      end
+
+      it "does not increment the resurrection counter when enable_resurrection_counter is not set" do
```
Comment (DenisSychyov) [inline spec/sidekiq/ultimate/resurrector_spec.rb]: Probably not `enable_resurrection_counter is not set`, but `enable_resurrection_counter proc returns 'false'`

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -103,9 +99,10 @@
   # as the one that triggered the failure.
   Kernel.srand config.seed
 
+  ENV["REDIS_NAMESPACE_QUIET"] = "1"
```
Comment (DenisSychyov) [inline spec/spec_helper.rb]: Let's maybe comment why we need that

File: spec/sidekiq/ultimate/resurrector_spec.rb (hunk 3 above)
Comment (sensorsasha) [reply spec/sidekiq/ultimate/resurrector_spec.rb]: https://github.com/sensortower/sidekiq-ultimate/pull/4/commits/888b899e03cd7717d1491ebe03cb5fb0d47d5eb5

File: spec/spec_helper.rb (hunk 4 above)
Comment (sensorsasha) [reply spec/spec_helper.rb]: https://github.com/sensortower/sidekiq-ultimate/pull/4/commits/08e0e5386d400ab0947a02f46c5216e40d081c92

File: README.md (hunk 1 above)
Comment (sensorsasha) [reply README.md]: https://github.com/sensortower/sidekiq-ultimate/pull/4/commits/ca423b1ec331aca31a49664e3471bd1a9bf2e66c

**Review by DenisSychyov — APPROVED** [review]

**Comment by kevinrobell-st** [comment]
When would you need to use a proc instead of a constant for the resurrection counter? I can't think of a case where that would be useful.

**Comment by sensorsasha** [comment]
> When would you need to use a proc instead of a constant for the resurrection counter? I can't think of a case where that would be useful.

When you want to enable/disable the resurrection counter in runtime without redeploying. For example, if you're afraid of an additional load on redis because of the resurrection counter.

**Comment by kevinrobell-st** [comment]
> > When would you need to use a proc instead of a constant for the resurrection counter? I can't think of a case where that would be useful.
> 
> When you want to enable/disable the resurrection counter in runtime without redeploying. For example, if you're afraid of an additional load on redis because of the resurrection counter.

Make sense from reading the updated README.

**Review by kevinrobell-st — APPROVED** [review]

**Pull request merged**